pkg/inst/context.go

```diff
@@ -3,7 +3,12 @@
 
 package inst
 
-type Context interface {
+// !!! Any modification MUST be synced with tool/internal/instrument/template_api.go
+type HookContext interface {
+	// Set the skip call flag, can be used to skip the original function call
+	SetSkipCall(bool)
+	// Get the skip call flag, can be used to skip the original function call
+	IsSkipCall() bool
 	// Set the data field, can be used to pass information between Before and After hooks
 	SetData(interface{})
```

Comment:
nit: We should migrate using `any` instead of `interface{}`.

Reply:
Let's stick to using interface{} in the generated code for better backward compatibility. This ensures the code will compile correctly in environments where any is not yet defined. In normal tool code, `any` is definitively better and preferable

Reply:
If we decide to keep the integrations within the same repo, believe me supporting old versions will be a lot problem. But this is another discussion.

There are linters and a modern go initiative as part of `go vet` will update these sort of patterns. So I don't want to make this a bikeshedding discussion :)